process/nodeStatusProcessor.go

```diff
@@ -17,28 +18,36 @@ const NetworkStatusPath = "/network/status"
 // NetworkConfigPath represents the path where an observer exposes his network metrics
 const NetworkConfigPath = "/network/config"
 
-// EconomicsDataPath represents the path where an observer exposes his economics data
-const EconomicsDataPath = "/network/economics"
-
 // NetworkConfigPath represents the path where an observer exposes his node status metrics
 const NodeStatusPath = "/node/status"
 
-// TotalStakedPath represents the path where an observer exposes the total staked value from validators contract
-const TotalStakedPath = "/network/total-staked"
-
 // NodeStatusProcessor handles the action needed for fetching data related to status metrics from nodes
 type NodeStatusProcessor struct {
-	proc Processor
+	proc                  Processor
+	economicMetricsCacher GenericApiResponseCacheHandler
+	cacheValidityDuration time.Duration
 }
 
 // NewNodeStatusProcessor creates a new instance of NodeStatusProcessor
-func NewNodeStatusProcessor(processor Processor) (*NodeStatusProcessor, error) {
+func NewNodeStatusProcessor(
+	processor Processor,
+	economicMetricsCacher GenericApiResponseCacheHandler,
+	cacheValidityDuration time.Duration,
+) (*NodeStatusProcessor, error) {
 	if check.IfNil(processor) {
 		return nil, ErrNilCoreProcessor
 	}
+	if check.IfNil(economicMetricsCacher) {
+		return nil, ErrNilEconomicMetricsCacher
+	}
+	if cacheValidityDuration <= 0 {
```

Comment:
could have been a larger value like 10 millisecond

Reply:
it is ok. it's basically a protection for default toml values